clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/models/output/VCVTrait.scala

```diff
@@ -0,0 +1,80 @@
+package org.broadinstitute.monster.etl.clinvar.models.output
+
+import io.circe.Encoder
+import io.circe.derivation.{deriveEncoder, renaming}
+import org.broadinstitute.monster.etl.MsgTransformations
+import org.broadinstitute.monster.etl.clinvar.ClinvarConstants
+import ujson.StringRenderer
+import upack.{Msg, Obj, Str}
+
+/**
+  * Info about a trait approved by ClinVar.
+  *
+  * @param id unique ID of the trait, corresponding to ClinVar's internal TraitID
+  * @param medgenTraitId unique ID of the trait in NCBI's MedGen database, if known
+  * @param name full name of the trait
+  * @param `type` type of the trait
+  */
+case class VCVTrait(
+  id: String,
+  medgenTraitId: Option[String],
+  name: Option[String],
+  `type`: Option[String]
+)
+
+object VCVTrait {
+  import org.broadinstitute.monster.etl.clinvar.MsgOps
+
+  implicit val encoder: Encoder[VCVTrait] = deriveEncoder(renaming.snakeCase, None)
+
+  /** Extract a VCVTrait from a raw Trait payload. */
+  def fromRawTrait(traitSet: VCVTraitSet, rawTrait: Msg): VCVTrait = {
+
+    val preferredNameArray = MsgTransformations
+      .getAsArray(rawTrait, "Name")
+      .filter(_.obj(Str("ElementValue")).obj(Str("@Type")).str == "Preferred")
+      .toArray
+
+    if (preferredNameArray.length > 1) {
+      throw new IllegalStateException(
+        s"VCV Trait in set ${traitSet.id} contains two Preferred names"
+      )
+    }
+
+    val allXrefs = MsgTransformations.popAsArray(rawTrait, "XRef")
+    val (medgenId, _) =
```

Comment:
This is throwing away the non-medgen xrefs. We're already implementing the processing needed for an `otherXrefs` field here, so it'd be low-hanging fruit to add that column. Otherwise we should leave non-medgen alone and merge them back into the content.